docs/src/main/asciidoc/redis-reference.adoc

```diff
@@ -370,7 +377,37 @@ The `canHandle` method is called to check if the codec can handle a specific typ
 The parameter received in the `encode` method matches that type.
 The object returned by the `decode` method must also match that type.
 
-=== The `value` group
+=== Use type reference
+
+Each group can be configured with `Class`, or with `TypeReference` objects.
+`TypeReference` are useful when dealing with Java generics:
+
+[source,java]
+----
+@ApplicationScoped
+public class MyRedisService {
+
+    private static final String MY_KEY = "my-key";
+
+    private final HashCommands<String, String, List<Person>> commands;
+
+    public MyRedisService(RedisDataSource ds) {
+        commands = ds.hash(new TypeReference<List<Person>>(){});
+    }
+
+    public void set(String field, List<Person> value) {
+        commands.hset(MY_KEY, field, value);
+    }
+
+    public List<Person> get(String field) {
+        return commands.hget(MY_KEY, field);
+    }
+}
+----
+
+IMPORTANT: You cannot use type references when using transaction. This is a known limitation.
```

Comment:
For my own knowledge, where is the limitation coming from?

Reply:
Time :-D 

It is technically possible to add support for type reference when using transactions. However, we never had that specific request.

Reply:
I thought so as well, thanks for the reply. Makes sense, and when the request comes, we can build on top of this PR.